### DISCUSSION TITLE
Assign Text to a non-text SVG element

### DISCUSSION THREAD

**Discussion by m90att** [ExperienceLovelace/ha-floorplan#265] [Q&A]
Hi,

Question 1 of 2 today :-)

Is it possible to assign text (from a sensor entity state) to a non-text svg element?  The reason I'm asking is because I've warped/skewed the element to display the text aligned to a window in my lounge.  To do so, in Inkscape I needed to convert the text to an object.  This splits the text letters into paths.  In inkscape there are 3 path objects forming the text group...

![image](https://user-images.githubusercontent.com/56609577/202686035-8673f5c2-7ac8-49b0-9541-f6a1085972e2.png)

If the element is a text element, it's easy to populate it with floorplan.text_set however this doesn't work now that the element is a (group of) paths.

Thanks

**Comment by exetico** [comment]
So, you're trying to add a text within a svg-DOM element, without a <tect> wrapper?
If so, the answer is most likely to be: Yes, it's possible, but outside the SVG-specs. You're breaking the standards by doing so.
_I'll not point you in that direction right now. If I remember correctly, this has been brought up in the past, but I can't find the discussion right now._

...But...

Please share what you mean by "align text to a window".

I don't get why you've accepted that Inkscaped converted it to svg-paths in the first place, instead of rotating the element or so, to be aligned with the window? :smiley: But I bet it makes more sense, if the SVG and a visual example was provided :smile_cat: !

**Reply by m90att** [reply]
Of course, an example would help...

This is my temperature display in my 'normal screen' svg...

![image](https://user-images.githubusercontent.com/56609577/202770278-2cb3f70c-cd7d-4315-a0b4-ee1b1f77e77d.png)

I'm working on a dedicated mobile svg and i've adjusted the orientation of the svg background.  This means that the window now looks like this (screenshot from inkscape)...

![image](https://user-images.githubusercontent.com/56609577/202770516-9ee80f96-21b1-4c46-9062-51059527610e.png)

You'll see that rather than a slant/skew it's more of a flute/funnel.  I'm unable to achieve that using the manipulation of the text box.  Googling showed me how to fit the text in a shape but that involves converting to an object, as mentioned in my original question.

Can you see any way of achieving this?

**Reply by exetico** [reply]
Thank you for the visuals. It would speed up the proces, if I had a small and **simplified** pack with both the SVG, CSS and YAML related to the question 😊.

**Reply by m90att** [reply]
Having continued to research this, I've concluded that it's not possible.  Non-text SVG elements are paths made up of co-ordinates so it's impossible to assign text characters to these.  I've used CSS transform to get them as close as I can to how I want them.  If I ever get time I might learn about CSS transform matrix as it looks like that may help with my situation.

**Reply by ferbulous** [reply]
Hi @m90att do you mind sharing your CSS code on warping/transforming the text for the sensors?

**Reply by OzGav** [reply]
@ferbulous You just transform/rotate the text of the object in Inkscape. Then when floorplan replaces the text it will look the same.

**Reply by ferbulous** [reply]
@OzGav rotate does work, but I'm trying to distort the text at odd angle similar to @m90att [attempted](https://github.com/ExperienceLovelace/ha-floorplan/discussions/265#discussioncomment-4179407) which I'm also not successful.
Was hoping for some example how to do it with CSS transform

**Reply by m90att** [reply]
I gave up in the end. I spent an inordinate amount of time trying to do it and my graphics skills just aren't up to the job.  I can't believe there's no intuitive way to create a trapezium/funnel.

**Reply by ferbulous** [reply]
I just remembered people have been using chatgpt to write automation/others, so I asked it to give me some css text transform example and this worked for me after modifying the numbers

```
.static-temp, .static-temp tspan {
	fill: #ffffff;
	position: fixed;
	transform: skew(213deg) scale(1.5) translateY(-190px) translateX(70px);
  }
```
<img width="548" alt="image" src="https://user-images.githubusercontent.com/24526737/210945558-6d6655e6-ce8f-4866-903b-ffb7c040d3de.png">

**Reply by OzGav** [reply]
Fantastic! I’m hearing chatgpt in every second conversation these days!
